libmscore/page.cpp

```diff
@@ -399,9 +407,11 @@ void PageFormat::read(XmlReader& e, Score* score)
                         }
                   }
             else if (tag == "page-height")
-                  _size.rheight() = e.readDouble() * 0.5 / PPI;
+                  _size.rheight() = e.readDouble() / SCALE_XML;
             else if (tag == "page-width")
-                  _size.rwidth() = e.readDouble() * .5 / PPI;
+                  _size.rwidth() = e.readDouble() / SCALE_XML;
+            else if (tag == "page-units")
```

Comment:
If you want to use a string, the obvious candidate is unitSuffixes[] itself, with "mm", "in", etc..
Not storing the page-units if the value is Units::MM or "mm" is a very easy adaptation.  I assume you still want millimeters as the default.

Note that regarding default settings for the Page Settings dialog, there was a comment about making the units a global user setting in the code here:
https://github.com/musescore/MuseScore/pull/2573/files#diff-b8d24849ea6144a41df80b460106035aL50
And Marc suggested the same thing here, and my response below it (#8) too:
https://musescore.org/en/node/108196#comment-488171

A totally separate future PR, if it is to be done, but worth noting in this context.